### PR TITLE
Split definition methods out of Broadcast Manager into new Definition Sender

### PR DESCRIPTION
**Pull request by awrichar**
Not totally crazy about the names `defsender.Sender` and `defhandler.DefinitionHandler`, but they need to be separate packages and this is where I landed...

### CODE REVIEW

File: internal/defsender/manager.go [hunk 1]
```diff
@@ -53,57 +88,36 @@ func (bm *broadcastManager) BroadcastIdentityClaim(ctx context.Context, ns strin
 	return bm.broadcastDefinitionCommon(ctx, ns, def, signingIdentity, tag, waitConfirm)
 }
 
-func (bm *broadcastManager) broadcastDefinitionCommon(ctx context.Context, ns string, def fftypes.Definition, signingIdentity *fftypes.SignerRef, tag string, waitConfirm bool) (*fftypes.Message, error) {
+func (bm *definitionSender) broadcastDefinitionCommon(ctx context.Context, ns string, def fftypes.Definition, signingIdentity *fftypes.SignerRef, tag string, waitConfirm bool) (*fftypes.Message, error) {
```
Comment (awrichar) [inline internal/defsender/manager.go:91]: Just want to note that this is probably the only method with functional changes beyond simple renaming. Due to the new separation between broadcast manager and definition sender, less of the data processing is done directly. We simply build a `MessageInOut` with `InlineData`, similar to any other broadcast. It feels clean overall, but I just wanted to point it out among the noise.

**Review by peterbroadhurst — COMMENTED** [review]
Just checking @awrichar - could `DefinitionSender` go into the `broadcast` package? (hence leaving the `definitions` package with its current name, but only containing the handlers)

**Comment by awrichar** [comment]
Yea, it probably could. But now I'm thinking it's best to pause on this until FIR-12 moves along a little further. The thought is that soon definitions will _not_ necessarily be a broadcast, so they don't totally belong in the broadcast package. Going to wait until that next piece of work queues up though before doing any more shuffling here.

**Comment by peterbroadhurst** [comment]
Converted to draft based on last comment

**Comment by awrichar** [comment]
Replaced by #878

**Pull request closed without merging**